### PR TITLE
ROX-23254: Email Sender add API package

### PR DESCRIPTION
**Pull request by kurlov**
## Description

- `handlers` should contain parsing request and generating response logic.
- `middleware` contains common wrappers for endpoint handlers
- `routes` contains path to handler mapping

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources
- [ ] (If applicable) Changes to the dp-terraform Helm values have been reflected in the addon on integration environment

## Test manual

```
CLUSTER_ID=test go run emailsender/cmd/app/main.go
```

### CODE REVIEW

File: emailsender/pkg/api/middleware.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"mime"
+	"net/http"
+
+	"github.com/golang/glog"
+)
+
+// LogRequest logs HTTP request
+func LogRequest(next http.Handler) http.Handler {
```
Comment (johannes94) [inline emailsender/pkg/api/middleware.go:11]: Did you consider reusing? If so why does it not fit? https://github.com/stackrox/acs-fleet-manager/blob/ad4a434c74938b4c18dcab1e3d436acee3d0d4bd/pkg/server/logging/request_logging_middleware.go#L13

File: emailsender/pkg/api/routes.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	"github.com/gorilla/mux"
+)
+
+// SetupRoutes configures API route mapping
+func SetupRoutes(router *mux.Router) {
+	// add middlewares
+	router.Use(LogRequest, EnsureJSONContentType)
+
+	router.HandleFunc("/health", HealthCheckHandler).Methods("GET")
+	router.HandleFunc("/v1/acscsemail", SendEmailHandler).Methods("POST")
```
Comment (johannes94) [inline emailsender/pkg/api/routes.go:13]: Do you want to additionally add the `/api` prefix, so this is consistent with other APIs?

Comment (kurlov) [reply emailsender/pkg/api/routes.go]: You are right, I also added `/rhacs`

Comment (johannes94) [reply emailsender/pkg/api/routes.go]: Ok, now `stackrox/stackrox` repo is again wrong. Should we fix it here to have `/api/v1/acscsemail` or change the integration one more time?

Comment (kurlov) [reply emailsender/pkg/api/routes.go]: Changed to `/api/v1/acscsemail`

**Review by johannes94 — APPROVED** [review]

**Pull request merged**